geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/operations/GetAndPutJsonDocumentsDUnitTest.java

```diff
@@ -0,0 +1,248 @@
+
+    // create a distributed region in two VMs so that we can store an object
+    // in "storingVM" and so ensure that it is in serialized form in the other
+    // VM
+    getCache().<String, Object>createRegionFactory(RegionShortcut.REPLICATE).create("TestRegion");
+
+    storingVM = Host.getHost(0).getVM(0);
+    storingVM.invoke("create region", () -> {
+      getCache().<String, Object>createRegionFactory(RegionShortcut.REPLICATE).create("TestRegion");
+    });
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    serializationService = null;
+    Cache cache = basicGetCache();
+    if (cache != null) {
+      cache.close();
+    }
+  }
+
+
+
+  @Test
+  public void testThatGetReturnsJSONDocumentForPdxInstance() throws Exception {
+    storeTestDocument();
+
+    RegionAPI.GetRequest getRequest = generateGetRequest(key);
+    GetRequestOperationHandler operationHandler = new GetRequestOperationHandler();
+    Result result = operationHandler.process(serializationService, getRequest,
+        TestExecutionContext.getNoAuthCacheExecutionContext(getCache()));
+
+    Assert.assertTrue(result instanceof Success);
+    RegionAPI.GetResponse response = (RegionAPI.GetResponse) result.getMessage();
+    assertEquals(BasicTypes.EncodedValue.ValueCase.JSONOBJECTRESULT,
+        response.getResult().getValueCase());
+    String actualValue = response.getResult().getJsonObjectResult();
+    assertEquals(jsonDocument, actualValue);
+  }
+
+  @Test
+  public void testThatGetAllReturnsJSONDocumentForPdxInstance() throws Exception {
+    storeTestDocument();
+
+    RegionAPI.GetAllRequest getRequest = generateGetAllRequest(key);
+    GetAllRequestOperationHandler operationHandler = new GetAllRequestOperationHandler();
+    Result result = operationHandler.process(serializationService, getRequest,
+        TestExecutionContext.getNoAuthCacheExecutionContext(getCache()));
+
+    Assert.assertTrue(result instanceof Success);
+    RegionAPI.GetAllResponse response = (RegionAPI.GetAllResponse) result.getMessage();
+    BasicTypes.Entry entry = response.getEntriesList().get(0);
+    BasicTypes.EncodedValue entryValue = entry.getValue();
+    assertEquals(BasicTypes.EncodedValue.ValueCase.JSONOBJECTRESULT, entryValue.getValueCase());
+    String actualValue = entryValue.getJsonObjectResult();
+    assertEquals(jsonDocument, actualValue);
+  }
+
+  @Test
+  public void testThatPutCreatesPdxInstanceFromJsonDocument() throws Exception {
+    RegionAPI.PutRequest putRequest = generatePutRequest(key, jsonDocument);
+    PutRequestOperationHandler operationHandler = new PutRequestOperationHandler();
+    Result result = operationHandler.process(serializationService, putRequest,
+        TestExecutionContext.getNoAuthCacheExecutionContext(getCache()));
+
+    Assert.assertTrue(result instanceof Success);
+    PdxInstance pdxInstance = (PdxInstance) getCache().getRegion(regionName).get(key);
+    assertEquals("__GEMFIRE_JSON", pdxInstance.getClassName());
```

Comment:
Is there a constant that could be used instead of the string literal?

Reply:
I looked for a constant but couldn't find one